docs/source/examples/dcgan.rst

```diff
@@ -0,0 +1,431 @@
+It is a kind of generative model with deep neural network, and often applied to
+the image generation. The GAN technique is also applied to 
+`PaintsChainer <https://paintschainer.preferred.tech/index_en.html>`_,
+a famous automatic coloring service.
+
+.. image:: ../../image/dcgan/generated-images.gif
+
+In the tutorial, you will learn about the following things:
+
+1. Basic Idea of Generative Model
+2. The Difference among GAN and Other Generative Models
+3. Details of GAN
+4. Implementation of DCGAN in Chainer
+
+1. Basic Idea of Generative Model
+==================================
+
+1.1 What is the Model?
+-----------------------
+
+In the field of science and engineering, we describe a system using mathematical
+concepts and language. The description is called as a **mathematical model**,
+and the process of developing a mathematical model is **mathematical modeling**.
+Especially, in the context of machine learning, we explain a target model by a
+map :math:`f` from an input :math:`x` to an output :math:`y`.
+
+.. math::
+    f: x \mapsto y
+    
+Therefore, the purpose of model training is obtaining the map :math:`f` from training data.
+In the case of unsupervised learning, we use the datasets of inputs
+:math:`\{s^{(n)}\}=\{d_1, d_2, \cdots, d_N\}` as the training data,
+and create model :math:`f`.
+In supervised learning, we use the datasets of inputs and their outputs
+:math:`\{s^{(n)}\}=\{(d_1, c_1), \cdots, (d_N, c_N)\}`. As a simple example,
+let's consider about a supervised learning problem such as classifying images as
+dogs or cats. Then, the training datasets consist of input images
+:math:`d_1, d_2, \cdots, d_N` and their labels
+:math:`c_1={\rm cat}, c_2={\rm dog}, \cdots, c_N={\rm cat}`.
+
+1.2 What is the Generative Model?
+----------------------------------
+
+When we consider about the generative model, it models the probability distribution
+:math:`p: s \mapsto p(s)` which generates the training data :math:`s`. The most simple
+generative model models the probability distribution :math:`p` with the map :math:`f`.
+We assign each :math:`x` and :math:`y` of :math:`f: x \mapsto y` as follows.
+
+* :math:`x` : the training data :math:`s`
+* :math:`y` : the likelihood of generating the training data :math:`s`
+
+In the case, because we models the probability distribution :math:`p` explicitly,
+we can calculate the likelihood :math:`p(s)`. So, we can maximize the likelihood.
+There is an advantage that the training process is simple. However, there is a
+disadvantage that we have to make a mechanism for sampling because we have only
+the process for calculating the likelihood.
+
+In the fitst place, we often just want to sample :math:`s \sim p(s)`
+according to the distribution in practice. The likelihood :math:`p(s)` is used
+only for model training. In the case, we sometimes do not model the probability distribution
+:math:`p(s)` directly, but other targets to facilitate sampling. 
+
```

Comment:
How about inserting this sentence here:
There are two approaches to model the probability distribution :math:\`p(s)\`.

Reply:
fixed.